.github/workflows/deploy.yml

```diff
@@ -0,0 +1,102 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - development
+      - base
+  workflow_dispatch:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/development'
+    environment:
+      name: ${{ github.ref_name }}
+
+    env:
+      APPLICATION_NAME: gemini
+      CONTAINER: gemini-web
+      APPLICATION_PORT: 8006
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Export secrets to environment variables
+        uses: oNaiPs/secrets-to-env-action@v1.5
+        with:
+          secrets: ${{ toJSON(secrets) }}
+
+      - name: Clone deploy scripts if not present
+        run: |
+          if [ ! -d deploy_scripts ]; then
+            git clone https://github.com/RockefellerArchiveCenter/deploy_scripts.git;
+          fi
+
+      - name: Substitute environment variables
+        uses: tvarohohlavy/inplace-envsubst-action@v1.0.0
+        with:
+          files: |
+            $APPLICATION_NAME/config.py.deploy
+            .env.deploy
+            appspec.yml.deploy
+            deploy_scripts/create_apache_config.sh.deploy
+            deploy_scripts/curl_index.sh.deploy
+            deploy_scripts/curl_status_endpoint.sh.deploy
+            deploy_scripts/install_dependencies_django.sh.deploy
+            deploy_scripts/restart_apachectl.sh.deploy
+            deploy_scripts/run_management_commands_django.sh.deploy
+            deploy_scripts/set_permissions.sh.deploy
+            deploy_scripts/stop_cron.sh.deploy
+
+      - name: Rename deploy files
+        run: |
+          mv $APPLICATION_NAME/config.py.deploy $APPLICATION_NAME/config.py
+          mv .env.deploy .env
+          mv appspec.yml.deploy appspec.yml
+          mv deploy_scripts/create_apache_config.sh.deploy deploy_scripts/create_apache_config.sh
+          mv deploy_scripts/curl_index.sh.deploy deploy_scripts/curl_index.sh
+          mv deploy_scripts/curl_status_endpoint.sh.deploy deploy_scripts/curl_status_endpoint.sh
+          mv deploy_scripts/install_dependencies_django.sh.deploy deploy_scripts/install_dependencies_django.sh
+          mv deploy_scripts/restart_apachectl.sh.deploy deploy_scripts/restart_apachectl.sh
+          mv deploy_scripts/run_management_commands_django.sh.deploy deploy_scripts/run_management_commands_django.sh
+          mv deploy_scripts/set_permissions.sh.deploy deploy_scripts/set_permissions.sh
+          mv deploy_scripts/stop_cron.sh.deploy deploy_scripts/stop_cron.sh
+
+      - name: Make deploy scripts executable
+        run: chmod +x deploy_scripts/*.sh
+
+      - name: Create deployment zip
+        run: sudo deploy_scripts/make_zip_django.sh $DEPLOY_ZIP_DIR $DEPLOY_ZIP_NAME
+
+      - name: Configure AWS Credentials
```

Comment:
Yeah, so this is a temporary workaround. The dev CodeDeploy role doesn't have permissions for prod CodeDeploy. 

As discussed, the ideal would be to have all dev resources (CodeDeploy) on the same account. I just went for the route of adding permissions to assume the prod role for CodeDeploy. I opted for the dev deployment role that already exists for the S3 push. 

Ideally, we'd just need a single role. I _could_ play with adding prod permissions to the dev CodeDeploy role if that's preferable.

Reply:
I'm just trying to think about making our lives easier in the future. We're going to have to update the role ARN anyway, but is there a way we can avoid having to update the workflow YAML as well?

Reply:
Ok, I just updated the Role ARN for dev to match permissions for the prod one.

Unfortunately, I had forgotten what the original issue was. As far as I can tell, there's no way to create a code deploy across different accounts because it won't find the application you need.

I don't see any way to specify which account to create a CodeDeploy deployment in the [documentation](https://github.com/RockefellerArchiveCenter/gemini/actions/runs/8993118588/job/24704293632).

You can see the error message [here](https://github.com/RockefellerArchiveCenter/gemini/actions/runs/8993118588/job/24704293632).

Reply:
OK great, as discussed we can continue to use the prod role, which has all the necessary permissions.